### PR TITLE
KodaDot: AssetHub NFT indexer M1

### PR DESCRIPTION
**Pull request by vikiival**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [ ] This pull request is being made by the same account as the accepted application. - Original was opened by @yangwao, we are both founders of KodaDot 
- [x] I have disclosed any and all sources of reused code in the submitted repositories and have done my due diligence to meet its license requirements.
- [x] In case of acceptance, an invoice must be submitted and the payment will be transferred to the Polkadot/fiat account provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://grants.web3.foundation/docs/Support%20Docs/milestone-deliverables-guidelines).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/1864 < please fill this in with the PR number of your application.

### CODE REVIEW

**Comment by yangwao** [comment]
> - [ ]  This pull request is being made by the same account as the accepted application. - Original was opened by @yangwao, we are both founders of KodaDot

Hello, @PieWol, should I open PR under my account, or is this one ok?

**Comment by PieWol** [comment]
> > * [ ]   This pull request is being made by the same account as the accepted application. - Original was opened by @yangwao, we are both founders of KodaDot
> 
> Hello, @PieWol, should I open PR under my account, or is this one ok?

Hey @vikiival this is fine :)

**Comment by PieWol** [comment]
Hey @vikiival ,
which branch should I use for testing, which readme to follow? arrow stick or main?

**Comment by vikiival** [comment]
> which branch should I use for testing

`arrow-stick` as linked in the readme

> Trying to "quickly running the sample" of the arrow-stick branch readme I get this erro

Yes there is `update db` missing, we forgot to add it to the readme:

```
npx squid-typeorm-migration apply
```

https://github.com/kodadot/stick/blob/5cdeec1139e3a68fa2ecec1427d4b9d02b2d152a/justfile#L47

https://docs.subsquid.io/sdk/troubleshooting/persisting-data/

I will add that to the readme

**Comment by PieWol** [comment]
@vikiival as I had this error on the "just process" step shouldn't the "just migrate" step be executed before "just process"? Right now it's the other way around in the readme. Good news is that it works now. Thanks for the quick fix :)

**Comment by vikiival** [comment]
> Right now it's the other way around in the readme

Will fix, thanks for the mention

**Comment by yangwao** [comment]
> Good news is that it works now. Thanks for the quick fix :)

Hey, excuse me for being pushy, but are there any updates from your side, @PieWol ? 
We need this to be approved, so we can merge few outstanding internal PRs for users experience on [kodadot.xyz](https://kodadot.xyz)
Namely
- https://github.com/kodadot/stick/pull/194
- https://github.com/kodadot/nft-gallery/pull/8920

Thank you for your swift reply ahead.

**Comment by PieWol** [comment]
Hey @yangwao sorry for the delay,
I'll have your evaluation ready tomorrow.

**Comment by PieWol** [comment]
Hey @vikiival ,
the testing deliverable stated the following.

> Testing will cover major functionality with unit tests and provide a guide for executing these tests.

Where do I find the testing guidelines and the unit tests? The current link doesn't point to a guide.

**Comment by vikiival** [comment]
> Hey @vikiival , the testing deliverable stated the following.
> 
> > Testing will cover major functionality with unit tests and provide a guide for executing these tests.
> 
> Where do I find the testing guidelines and the unit tests? The current link doesn't point to a guide.

Hello,
[we follow the classical convention of running the tests](https://github.com/kodadot/stick/blob/1cda35bd2cba5dc20aa37bd9005d2158e1aeb497/package.json#L11)

[other tests are part of hyperdata](https://github.com/kodadot/packages/tree/main/hyperdata/tests)

TL;DR all tests are located in the `tests` directory and to run them one must use `npm run test`

**Comment by PieWol** [comment]
Hey @vikiival this is my current [evaluation](https://github.com/w3f/Grant-Milestone-Delivery/pull/1133) for this project. I have a few questions in the notes and listed things that should be added. Looking forward to your response and changes.

File: deliveries/Kodadot_assethub_nft_indexer_1.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+| 8b.    | Implementing Metadata Caching Layer  | [retry IPFS metadata if un-indexed by Metadata IPFS Integration Handler.](https://github.com/kodadot/stick/blob/arrow-stick/src/mappings/utils/cache.ts)                                             | -                                                                                    |
+| 9.     | Transfer of Collection Ownership     | [transfer collection ownership.](https://github.com/kodadot/stick/blob/arrow-stick/src/mappings/nfts/setAttribute.ts)                                                                   | -                                                                                    |
+| 10a.   | Collection settings Schema Design    | [GraphQL schema collection settrings.](https://github.com/kodadot/stick/blob/arrow-stick/schema.graphql#K93)                                            | -                                                                                    |
```
Comment (PieWol) [inline deliveries/Kodadot_assethub_nft_indexer_1.md:58]:
```suggestion
| 10a.   | Collection settings Schema Design    | [GraphQL schema collection settrings.](https://github.com/kodadot/stick/blob/arrow-stick/schema.graphql#L93)                                            | -                                                                                    |
```

**Comment by PieWol** [comment]
> @PieWol can we proceed? 🥺

As I said, team needs to review and merge. I can't speed up the process for you at this point. Please give us a little time.

**Review by semuelle — APPROVED** [review]
Thanks, @vikiival & @yangwao. I just confirmed the evaluation. Your milestone is hereby approved.

**Comment by vikiival** [comment]
Hey @semuelle I have already sent invoice on 30.01.2024.
Can we proceed with that one or should I make a new one?

Guess @keeganquigley was replying me.
